File: mambo-protocol/src/main/scala/org/mambo/protocol/types/game/look/SubEntity.scala





















// Generated on 05/03/2013 10:35:07
// Scala version by Blackrush
package org.mambo.protocol.types

import org.mambo.core.io._
import org.mambo.protocol.{Type => BaseType, _}
import org.mambo.protocol.enums._

trait SubEntity extends TypeDeserializer {
  val typeId = 54

  protected trait Type extends BaseType with Serializable {
    val typeId = SubEntity.typeId

    val bindingPointCategory: Byte
    val bindingPointIndex: Byte
    val subEntityLook: EntityLook.TypeImpl

    def serialize(buf: Buffer) {
      buf.writeByte(bindingPointCategory)
      buf.writeByte(bindingPointIndex)
      subEntityLook.serialize(buf)
    }

  }

  protected def create(bindingPointCategory: Byte, bindingPointIndex: Byte, subEntityLook: EntityLook.TypeImpl): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val bindingPointCategory = buf.readByte()
    val bindingPointIndex = buf.readByte()
    val subEntityLook = EntityLook.deserialize(buf)

    create(bindingPointCategory, bindingPointIndex, subEntityLook)
  }

}

object SubEntity extends SubEntity {
  case class TypeImpl(bindingPointCategory: Byte, bindingPointIndex: Byte, subEntityLook: EntityLook.TypeImpl) extends Type
  override def create(bindingPointCategory: Byte, bindingPointIndex: Byte, subEntityLook: EntityLook.TypeImpl) = TypeImpl(bindingPointCategory, bindingPointIndex, subEntityLook)

  def unapply(msg: TypeImpl) = TypeImpl.unapply(msg)
  def apply(bindingPointCategory: Byte, bindingPointIndex: Byte, subEntityLook: EntityLook.TypeImpl) = create(bindingPointCategory, bindingPointIndex, subEntityLook)

  type Target = TypeImpl

}

File: mambo-protocol/src/main/java/org/mambo/protocol/types/SubEntity.java


// Generated on 05/08/2013 19:38:03
package org.mambo.protocol.types;

import java.util.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class SubEntity extends NetworkType {
    public static final short TYPE_ID = 54;
    
    @Override
    public short getTypeId() {
        return TYPE_ID;
    }
    
    public byte bindingPointCategory;
    public byte bindingPointIndex;
    public EntityLook subEntityLook;
    
    public SubEntity() { }
    
    public SubEntity(byte bindingPointCategory, byte bindingPointIndex, EntityLook subEntityLook) {
        this.bindingPointCategory = bindingPointCategory;
        this.bindingPointIndex = bindingPointIndex;
        this.subEntityLook = subEntityLook;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeByte(bindingPointCategory);
        buf.writeByte(bindingPointIndex);
        subEntityLook.serialize(buf);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        bindingPointCategory = buf.readByte();
        if (bindingPointCategory < 0)
            throw new RuntimeException("Forbidden value on bindingPointCategory = " + bindingPointCategory + ", it doesn't respect the following condition : bindingPointCategory < 0");
        bindingPointIndex = buf.readByte();
        if (bindingPointIndex < 0)
            throw new RuntimeException("Forbidden value on bindingPointIndex = " + bindingPointIndex + ", it doesn't respect the following condition : bindingPointIndex < 0");
        subEntityLook = new EntityLook();
        subEntityLook.deserialize(buf);
    }
    
}
